BiasAndFairnessModule/src/visualizations/plots.py

```
@@ -0,0 +1,775 @@
+
+        # Prepare text annotations
+        if normalize is None:
+            text_vals = [[f"{int(v)}" for v in row] for row in cm]
+        else:
+            text_vals = [[f"{v:.2f}" for v in row] for row in cm]
+
+        # Heatmap trace
+        fig.add_trace(
+            go.Heatmap(
+                z=cm,
+                x=[str(l) for l in display_labels],
```

Ruff (0.12.2)

Ambiguous variable name: `l`

(E741)